utils/comparable_bytes.cc

```diff
@@ -527,11 +535,53 @@ struct to_comparable_bytes_visitor {
         out.write<uint8_t>(DECIMAL_LAST_BYTE);
     }
 
-    // TODO: Handle other types
-
+    // Encode variable length natively byte-comparable data types.
+    // Zeros in the source are escaped as ESCAPE followed by one or more ESCAPED_0_CONT and ending with ESCAPED_0_DONE.
+    // The encoding terminates in an escaped state (either with ESCAPE or ESCAPED_0_CONT).
+    // If the source ends in a zero, we use ESCAPED_0_CONT to ensure that the encoding remains smaller than
+    // the same source with an additional zero at the end.
+    //
+    // E.g. "A\0\0B" translates to 4100FEFF4200
+    //      "A\0B\0"               4100FF4200FE
+    //      "A\0"                  4100FE
+    //      "AB"                   414200
+    //
+    // In a single-byte source, the bytes could be passed unchanged, but this would prevent combining components.
+    // This translation preserves order, and since the encoding for 0 is higher than the separator,
+    // it also ensures that shorter components are treated as smaller.
+    //
+    // The encoding is not prefix-free since, for example, the encoding of "A" (4100) is a prefix of the encoding of "A\0" (4100FE).
+    // However, the byte following the prefix is guaranteed to be FE or FF, making the encoding weakly prefix-free.
+    // Additionally, any such prefix sequence will compare as smaller than the value it prefixes,
+    // because any permitted separator byte will be smaller than the byte following the prefix.
     void operator()(const abstract_type& type) {
-        // Unimplemented
-        on_internal_error(cblogger, fmt::format("byte comparable format not supported for type {}", type.name()));
+        if (!type.is_byte_order_equal()) {
+            on_internal_error(cblogger, fmt::format("byte comparable format not supported for type {}", type.name()));
+        }
+
+        bool escaped = false;
+        while (!serialized_bytes_view.empty()) {
+            uint8_t curr_byte = read_simple<uint8_t>(serialized_bytes_view);
+            if (escaped) {
```

Comment:
text is extremely common and also extremely unlikely to have zeros. So best have a fast-path that checks no zeros are there and just copy stuff, and a slow-path that operates byte-by-byte.

Also, loop on every fragment instead of every byte.